src/attribution/onFCP.ts

```diff
@@ -19,42 +19,40 @@ import {getLoadState} from '../lib/getLoadState.js';
 import {getNavigationEntry} from '../lib/getNavigationEntry.js';
 import {isInvalidTimestamp} from '../lib/isInvalidTimestamp.js';
 import {onFCP as unattributedOnFCP} from '../onFCP.js';
-import {
-  FCPMetric,
-  FCPMetricWithAttribution,
-  FCPReportCallback,
-  FCPReportCallbackWithAttribution,
-  ReportOpts,
-} from '../types.js';
+import {FCPMetric, FCPMetricWithAttribution, ReportOpts} from '../types.js';
+
+const attributeFCP = (metric: FCPMetric): FCPMetricWithAttribution => {
+  const metricWithAttribution = metric as FCPMetricWithAttribution;
```

Comment:
It does have the same problem, just keeps the type lies highly localized and so in theory easier to keep an eye on :)

Your proposal sounds good to make that even better. The early returns of a value (instead of undefined) were sometimes awkward for some of these anyways, so better all around.